CommonTools/Utils/src/FormulaEvaluator.cc

```diff
@@ -236,9 +236,10 @@ namespace {
       return false;
     }
 
-    EvaluatorInfo createEvaluator(std::string::const_iterator iBegin,
-                                  std::string::const_iterator iEnd,
-                                  std::shared_ptr<reco::formula::BinaryOperatorEvaluatorBase> iPreviousBinary) const {
+    EvaluatorInfo createEvaluator(
+        std::string::const_iterator iBegin,
+        std::string::const_iterator iEnd,
+        const std::shared_ptr<reco::formula::BinaryOperatorEvaluatorBase>& iPreviousBinary) const {
```

Comment:
not sure this is desirable - maybe it's better to modify this function to take the underlying pointer directly, if it's not supposed to have ownership?